missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java

```diff
@@ -657,10 +675,9 @@ public void setArena(Arena arena) {
         }
 
         this.arena = arena.clone();
-
-        // Load world
-        this.gameWorld = new GameWorld(this, this.arena.getTemplateWorld());
-        this.gameWorld.load();
+        gameWorld = new GameWorld(this, arena.getTemplateWorld());
+        gameWorld.load();
+        gameArea = new GameArea(gameWorld.getWorld(), arena.getAreaConfig());
 
```

Comment:
With `arena.getTemplateWorld()` and `arena.getAreaConfig()`, it doesn't matter whether you take `arena` or `this.arena`. One is a string, the other returns a new independent object. And `this.gameworld` is the same as `gameworld`.